src/app/settings/admin/rule-groups/component.ts

```diff
@@ -0,0 +1,200 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, OnDestroy, OnInit, OnChanges, ViewChild} from '@angular/core';
+import {MatDialog, MatDialogConfig} from '@angular/material/dialog';
+import {MatPaginator} from '@angular/material/paginator';
+import {MatSort} from '@angular/material/sort';
+import {MatTableDataSource} from '@angular/material/table';
+import {NotificationService} from '@core/services/notification';
+import {DatacenterService} from '@core/services/datacenter';
+import {MLAService} from '@core/services/mla';
+import {UserService} from '@core/services/user';
+import {ConfirmationDialogComponent} from '@shared/components/confirmation-dialog/component';
+import {RuleGroup, RuleGroupType, AdminRuleGroup} from '@shared/entity/mla';
+import {MLAUtils} from '@shared/utils/mla-utils';
+import {UserSettings} from '@shared/entity/settings';
+import _ from 'lodash';
+import {Subject} from 'rxjs';
+import {filter, switchMap, take, takeUntil} from 'rxjs/operators';
+import {Mode, AdminRuleGroupDialog} from './rule-group-dialog/component';
+
+@Component({
+  selector: 'km-admin-settings-rule-groups',
+  styleUrls: ['style.scss'],
+  templateUrl: 'template.html',
+})
+export class AdminSettingsRuleGroupsComponent implements OnInit, OnChanges, OnDestroy {
+  dataSource = new MatTableDataSource<AdminRuleGroup>();
+  displayedColumns: string[] = ['name', 'type', 'seed', 'actions'];
+  seeds: string[] = [];
+  seedFilter: string;
+  typeFilter: string;
+  settings: UserSettings;
+  ruleGroupTypes = Object.values(RuleGroupType);
+  adminRuleGroups: AdminRuleGroup[] = [];
+
+  @ViewChild(MatSort, {static: true}) sort: MatSort;
+  @ViewChild(MatPaginator, {static: true}) paginator: MatPaginator;
+
+  private _ruleGroupsMap = new Map<string, RuleGroup[]>();
+  private _unsubscribe = new Subject<void>();
+
+  constructor(
+    private readonly _matDialog: MatDialog,
+    private readonly _mlaService: MLAService,
+    private readonly _userService: UserService,
+    private readonly _notificationService: NotificationService,
+    private readonly _datacenterService: DatacenterService
+  ) {}
+
+  ngOnInit(): void {
+    this.dataSource.data = this.adminRuleGroups || [];
+    this.dataSource.sort = this.sort;
+    this.dataSource.paginator = this.paginator;
+    this.sort.active = 'name';
+    this.sort.direction = 'asc';
+
+    this._userService.currentUserSettings.pipe(takeUntil(this._unsubscribe)).subscribe(settings => {
+      this.settings = settings;
+      this.paginator.pageSize = settings.itemsPerPage;
+      this.dataSource.paginator = this.paginator; // Force refresh.
+    });
+
+    this._datacenterService.seeds.pipe(takeUntil(this._unsubscribe)).subscribe(seeds => {
+      this.seeds = seeds;
+      this._getAdminRuleGroups();
```

Comment:
This change creates a new issue. It will cause a subscription leak and create lots of the same subscriptions every time seeds get updated.

A pipe should be used here possibly with a `combineLatest` to control the rule groups.

Reply:
Understood the problem, but unfortunately my attemps are failing atm. Eather code is not correct or it makes it even worse, but no idea how to fix it. :/ Could you help out, please?

Reply:
Optimized